build_projects/dotnet-host-build/CompileTargets.cs

```diff
@@ -322,6 +323,107 @@ public static BuildTargetResult CompileCoreHost(BuildTargetContext c)
+
+            // Run the build
+            string uwpSrcDir = Path.Combine(c.BuildContext.BuildDirectory, "src", "uwp");
+
+            // Create .rc files
+            var resourceDir = GenerateVersionResource(c);
+
+            if (configuration.Equals("Release"))
+            {
+                // Cmake calls it "RelWithDebInfo" in the generated MSBuild
+                configuration = "RelWithDebInfo";
+            }
+
+            List<string> cmakeArgList = new List<string>();
+
+            string visualStudio, archMacro, arch;
+            string cmakeResourceDir = $"-DCLI_CMAKE_RESOURCE_DIR:STRING={resourceDir}";
+            string cmakeInstallPrefix = $"-DCMAKE_INSTALL_PREFIX:STRING={cmakeInstallPrefixDir}";
+            string cmakeExtraArgs = null;
+
+            switch (platform.ToLower())
+            {
+                case "x86":
+                    visualStudio = "Visual Studio 14 2015";
+                    archMacro = "-DCLI_CMAKE_PLATFORM_ARCH_I386=1";
+                    cmakeExtraArgs = "-DCMAKE_SYSTEM_VERSION=10.0";
+                    arch = "x86";
+                    break;
+                case "arm":
+                    visualStudio = "Visual Studio 14 2015 ARM";
+                    archMacro = "-DCLI_CMAKE_PLATFORM_ARCH_ARM=1";
+                    cmakeExtraArgs = "-DCMAKE_SYSTEM_VERSION=10.0";
+                    arch = "arm";
+                    break;
+                case "arm64":
+                    visualStudio = "Visual Studio 14 2015 Win64";
+                    archMacro = "-DCLI_CMAKE_PLATFORM_ARCH_ARM64=1";
+                    cmakeExtraArgs = "-DCMAKE_SYSTEM_VERSION=10.0";
+                    arch = "arm64";
+                    if (Environment.GetEnvironmentVariable("__ToolsetDir") == null)
+                    {
+                        throw new Exception("Toolset Dir must be set when the Platform is ARM64");
+                    }
+                    break;
+                case "x64":
+                    visualStudio = "Visual Studio 14 2015 Win64";
+                    archMacro = "-DCLI_CMAKE_PLATFORM_ARCH_AMD64=1";
+                    cmakeExtraArgs = "-DCMAKE_SYSTEM_VERSION=10.0";
+                    arch = "x64";
+                    break;
+                default:
+                    throw new PlatformNotSupportedException("Target Architecture: " + platform + " is not currently supported.");
+            }
+
+            cmakeArgList.Add(cmakeInstallPrefix);
+            cmakeArgList.Add(uwpSrcDir);
+            cmakeArgList.Add(archMacro);
+            cmakeArgList.Add(cmakeResourceDir);
+            cmakeArgList.Add("-G");
+            cmakeArgList.Add(visualStudio);
+
+            if (!String.IsNullOrEmpty(cmakeExtraArgs))
+            {
+                cmakeArgList.Add(cmakeExtraArgs);
+            }
+
+            ExecIn(cmakeOut, "cmake", cmakeArgList);
+
+            var pf32 = RuntimeInformation.OSArchitecture == Architecture.X64 ?
```

Comment:
I'd change the variable name to something more descriptive, and change the type to "string" since it's obvious that this is a string.  Change the other "vars" to "strings" as well for that matter.